src/chapters/typescript/classes.rst

```diff
@@ -0,0 +1,99 @@
+Classes and Interfaces in TypeScript
+====================================
+
+Classes in TypeScript look something like this:
+
+.. sourcecode:: js
+   :linenos:
+
+   class Astronaut {
+      name: string;
+      constructor(firstName: string, lastName: string) {
+         this.name = firstName + " " + lastName;
+      }
+      greet() {
+         return "Hello, " + this.name;
+      }
+   }
+
+   let Bob = Astronaut("Bob","Smith");
+
+You may remember the ``this`` and ``new`` keywords from working with classes in JavaScript.
+Earlier in the chapter, we also noted that when declaring variables in TypeScript, we have to specify the type of value.
+The same applies to function parameters, as you can see in the constructor.
+
+Classes in TypeScript can also use the ``extends`` keyword to denote child and parent classes, as shown here:
+
+.. sourcecode:: js
+   :linenos:
+
+   class Panthera {
+      roar: string;
+      constructor(currentRoar: string) {
+         this.roar = currentRoar;
+      }
+   }
+
+   class Tiger extends Panthera {
+      stripes: boolean = true;
+
+   }
+
+   let tigger = new Tiger("loud");
+   console.log(tigger.roar);
+   console.log(tigger.stripes);
+
+
+``export`` is used to make the class available in the global scope.
+
+Interfaces
+----------
+
+When we start working with Angular, you may see the ``interface`` keyword quite a bit. Interfaces in TypeScript name types that are created by the programmer.
+
+We may create an interface for a data type that contains all of the information we need about an astronaut and then use that information in a function.
+
+.. sourcecode:: js
+
+   interface Astronaut {
+      name: string;
+   }
+
+   function astronautName (astronaut: Astronaut) {
+      return astronaut.name;
+   }
+
+   let bob = {name: "Bob"};
+   console.log(astronautName(bob));
+
+
+``null`` and ``undefined`` are primitive data types in TypeScript, however, they are treated differently by TypeScript.
```

Comment:
I made it its own section. I didn't want to move it to a separate page, because I only brought it up in the context of optional parameters. Let me know what you think!